File: TareaHerencia/Solucion/Java/Villano.java
public class Villano extends Persona{
    public String nickname;
    public String poder;

    public Villano(String name, int age, String nick, String power){
        super(name,age);
        this.nickname = nick;
        this.poder = power;
    }

    public void presentacion(){
        System.out.println("Hola soy " + this.nickname + " y mi poder es " + this.poder);
    }

    public void revelarNombre(){
        System.out.println("Mi verdadero nombre es " + this.darNombre());
    }
}

File: TareaHerencia/Solucion/C++/Villano.cpp
#include "Villano.h"

Villano::Villano(string name, int age, string nick, string power) : Persona(name,age){
    nickname = nick;
    super_poder = power;
}

void Villano::presentacion(){
    cout<<"Hola soy "<<nickname<<" y mi poder es "<<super_poder<<endl;
}

void Villano::revelarNombre(){
    cout<<"Mi verdadero nombre es "<<darNombre()<<endl;
}